### ISSUE TITLE
Cannot read properties of undefined (reading 'apply') 

### ISSUE THREAD

**Issue by michaelAtCoalesce** [DataDog/browser-sdk#1570]
**Cannot read properties of undefined (reading 'apply')** 

TypeError: Cannot read properties of undefined (reading 'apply')```

the code that fails is clearly in the datadog RUM 

../../src/tools/instrumentMethod.ts::stop:17 
../../src/browser/xhrProxy.ts:(XMLHttpRequest.onreadystatechange::119 

```

This caused a failure on load for our application

        **"@datadog/browser-rum": "^4.7.1",**

**Comment by BenoitZugmeyer** [comment]
Hello @michaelAtCoalesce, thanks for the report.

The file you are indicating, `xhrProxy.ts`, does not exist since version `3.6.12`. Also it didn't use `instrumentMethod` back then. So I suspect something is wrong either with the stack trace or the version you are providing.

Could you provide additional details? A small example to reproduce the issue would be great!

**Comment by Ryankey** [comment]
I've recently come into what I think is the same issue on "^4.14.0". In our case, we have Datadog RUM, Sentry (for error watching) and a third library molstar combining to produce this error on only one part of our application.

The order of events is...

1. Molstar makes an initial xhr request
2. Datadog RUM wraps the `xhr.onreadystatechange` (which is currently null, as many callbacks are) on [L13](https://github.com/DataDog/browser-sdk/blob/ed8a9bfdb21e4067c795f81804617b3ddf121408/packages/core/src/tools/instrumentMethod.ts#L13). This wrapping is good and will succeed on our first request because the wrapping function handles non-functions on [L55](https://github.com/DataDog/browser-sdk/blob/ed8a9bfdb21e4067c795f81804617b3ddf121408/packages/core/src/tools/instrumentMethod.ts#L55)
3. Sentry wraps `xhr.onreadystatechange` for error watching
4. `stop()` is called on the wrapper from 2, but because of Sentry's wrap from 3 we fail the condition on [L23](https://github.com/DataDog/browser-sdk/blob/ed8a9bfdb21e4067c795f81804617b3ddf121408/packages/core/src/tools/instrumentMethod.ts#L23), which causes us to fall back to L26, replacing our safe monitoring wrapper with the original null
5. Molstar makes another xhr request, and uses a pool of xhr objects, reusing the previous one
6. We re-wrap the sentry wrapper and some other shenanigans, but eventually call [L17](https://github.com/DataDog/browser-sdk/blob/ed8a9bfdb21e4067c795f81804617b3ddf121408/packages/core/src/tools/instrumentMethod.ts#L17) with `instrumentation` which is now `null`, generating an exception `TypeError: Cannot read properties of null (reading 'apply')`

Not sure how to generate a reproducible example, but hopefully that provides enough information.

**Comment by BenoitZugmeyer** [comment]
Thank you very much for your investigation, that's very clear! I'll look at your PR shortly

**Comment by bcaudan** [comment]
👋 The fix is available in version 4.18.1, don't hesitate to reach out if there is still something wrong.
